app/controllers/rules_controller.rb

```diff
@@ -7,9 +7,12 @@ def index
   end
 
   def show
-    rule = ::Pundit.policy_scope(User.current, ::Rule).friendly.find(
-      params[:id]
-    )
+    rule = ::Pundit.policy_scope(User.current, ::Rule).where(
+      'rules.slug LIKE ?',
+      "%#{ActiveRecord::Base.sanitize_sql_like(params[:id])}%"
+    ).first
+    raise ActiveRecord::RecordNotFound if rule.blank?
```

Comment:
IMO `show` must guarantee uniqueness.

```suggestion
    rules = ::Pundit.policy_scope(User.current, ::Rule).where(
      'rules.slug LIKE ?',
      "%#{ActiveRecord::Base.sanitize_sql_like(params[:id])}%"
    )
    raise ActiveRecord::RecordNotFound unless rules.one?
```